### PR TITLE
Check dependent options for bad combinations

### PR DESCRIPTION
**Pull request by davidbeckingsale**
This PR checks for cases where `RAJA_ENABLE_X=On` but `ENABLE_X=Off` as this can be confusing since feature `X` will not be enabled despite the value of the RAJA specific option. Example output:

```
$ cmake -DRAJA_ENABLE_CUDA=On ..

...

CMake Error at cmake/SetupDependentOptions.cmake:19 (message):
  RAJA_ENABLE_CUDA set to On, but ENABLE_CUDA is Off.  Please set ENABLE_CUDA
  to On enable this feature.
Call Stack (most recent call first):
  CMakeLists.txt:110 (include)
```

and when `RAJA_ALLOW_INCONSISTENT_OPTIONS=On`:

```
$ cmake -DRAJA_ENABLE_CUDA=On -DRAJA_ALLOW_INCONSISTENT_OPTIONS=On ..

...
CMake Warning at cmake/SetupDependentOptions.cmake:17 (message):
  RAJA_ENABLE_CUDA set to On, but ENABLE_CUDA is Off.  Please set ENABLE_CUDA
  to On to enable this feature.
Call Stack (most recent call first):
  CMakeLists.txt:110 (include)
```

### CODE REVIEW

File: cmake/SetupDependentOptions.cmake [hunk 1]
```diff
@@ -9,6 +9,14 @@
 ## Here are the CMake dependent options in RAJA.
 ##
 
+set(RAJA_DEPENDENT_OPTIONS ENABLE_OPENMP ENABLE_CUDA ENABLE_HIP ENABLE_CLANG_CUDA ENABLE_COVERAGE ENABLE_TESTS ENABLE_EXAMPLES ENABLE_BENCHMARKS)
+foreach (option ${RAJA_DEPENDENT_OPTIONS})
+   if (${RAJA_${option}})
+      if (NOT ${option})
+         message(WARNING "RAJA_${option} set to On, but ${option} is Off")
```
Comment (rhornung67) [inline cmake/SetupDependentOptions.cmake:16]: Maybe beef up the message by adding something about ${option} must be on to enable feature??

Comment (davidbeckingsale) [reply cmake/SetupDependentOptions.cmake:16]: That's a good idea - so change the `WARNING` to `FATAL_ERROR` unless `RAJA_ALLOW_INCONSISTENT_OPTIONS` is set.

What do we want the default value for that option to be?

File: cmake/SetupDependentOptions.cmake [hunk 2]
```diff
@@ -13,7 +13,11 @@ set(RAJA_DEPENDENT_OPTIONS ENABLE_OPENMP ENABLE_CUDA ENABLE_HIP ENABLE_CLANG_CUD
 foreach (option ${RAJA_DEPENDENT_OPTIONS})
    if (${RAJA_${option}})
       if (NOT ${option})
-         message(WARNING "RAJA_${option} set to On, but ${option} is Off")
+         if (RAJA_ALLOW_INCONSISTENT_OPTIONS)
```
Comment (rhornung67) [inline cmake/SetupDependentOptions.cmake:16]: Are we still thinking that it may be good to push this feature into BLT?

Comment (rhornung67) [reply cmake/SetupDependentOptions.cmake:16]: We should discuss with Chris W.

**Review by rhornung67 — APPROVED** [review]

**Pull request merged**